daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -576,8 +576,9 @@ object Speedy {
     }
   }
 
-  /** The scrutinee of a match has been evaluated, now match the alternatives against it. */
-  final case class KMatch(alts: Array[SCaseAlt]) extends Kont with SomeArrayEquals {
+  /** The scrutinee of a match has been evaluated, now match the alternatives against it.
+    * Consult the documentation of `SECase` for the meaning of `jumpable`. */
+  final case class KMatch(alts: Array[SCaseAlt], jumpable: Boolean) extends Kont with SomeArrayEquals {
```

Comment:
Why not create specialize version of `KMacth`  
why not specializing  `KMach` for each matchable type ? 

```[scala]
final case class KMatchBoolean(ifCase: SExpr, elseCase: SExpr) extends Kont with SomeArrayEquals {
    override def execute(v: SValue, machine: Machine): Unit =
      machine.ctrl = CtrlExpr(if (v.asInstanceOf[SBool].value) ifCase else elseCase)
  }
  
  final case class KMatchEnum(alts: Array[SExpr]) extends Kont with SomeArrayEquals {
    override def execute(v: SValue, machine: Machine): Unit =
      machine.ctrl = CtrlExpr(alts(v.asInstanceOf[SEnum].constructorRank))
  }

  final case class KMatchVariant(alts: Array[SExpr]) extends Kont with SomeArrayEquals {
    override def execute(v: SValue, machine: Machine): Unit = {
      val variant = v.asInstanceOf[SVariant]
      machine.kont.add(KPop(1))
      machine.env.add(variant.value)
      machine.ctrl = CtrlExpr(alts(variant.constructorRank))
    }
  }
  ...

 ```

Reply:
I'm working on something similar in a more general fashion.